gradle-module-plugin/build.gradle.kts

```diff
@@ -41,6 +42,12 @@ val functionalTest by tasks.registering(Test::class) {
 
     group = "verification"
     description = "Executes tests in the 'functionalTest' sourceset"
+
+    // because this is a custom task or some other Gradle voodoo we can not rely on the debug system prop
```

Comment:
that's funky.  I wonder if this would be resolved correctly using [test suites](https://docs.gradle.org/current/userguide/jvm_test_suite_plugin.html) for configuring this sourceSet?  Prob out of scope for this PR, but might be worth it's own effort.

Reply:
Yeah I was thinking the same thing but. Also I have in the past seen other custom tasks based on well-known task classes not take on all the properties of those tasks. IIRC at the last shop I had a few that did not template the classpath, for example.